### PR TITLE
Don't write files that aren't changed, other improvements to XamlStyler.Console

### LINKED ISSUE #289: XamlStyler.Console changes file without... changes
**Describe the bug**
When I run the XamlStyler.Console dotnet tool on a folder, it processes all files, makes no change, because everything is OK, but changes the file in such a way that git adds modified file but without changes.

**To Reproduce**
Run the tool on a folder that is already formatted correctly.

**Expected behavior**
No file changes, and no git modified files.

### LINKED ISSUE #378: Update to .NET 6
**Is your feature request related to a problem? Please describe.**
.NET Core 3.1 will be EOL at the end of the year: https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

It seems like I can't run the .NET Core 3.1 based app in my .NET 6 based pipeline:
```
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '3.1.0' (x64) was not found.
  - The following frameworks were found:
      6.0.3 at [C:\Users\runneradmin\AppData\Local\Microsoft\dotnet\shared\Microsoft.NETCore.App]
```

(without also having to install .NET Core 3.1 alongside the .NET 6 SDK install.)

**Describe the solution you'd like**
It'd be great to update to use the `net6.0` TFM over `netcoreapp3.1`

### PR DESCRIPTION
**Pull request by michael-hawker**
### Description:

General improvements to XamlStyler.Console, updated dependencies, move to .NET 6, resolve issue with tool writing files that don't need to be changed, and add status code return on error when files/directories not specified.

Fixes #378 
Fixes #289

Wanted to test how this works in the CI, probably some things that need tweaking because I'm unfamiliar with your processes.

Happy to split this up into two PRs, or feel free to modify as needed.

### Checklist:
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] My changes generate no new warnings
* [ ] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [ ] I have tested my changes by running the extension in VS2017
* [ ] I have tested my changes by running the extension in VS2019
* [ ] I have tested my changes by running the extension in VS2022
* [ ] If changes to the [documentation](https://github.com/Xavalon/XamlStyler/wiki) are needed, I have noted this in the description above

### CODE REVIEW

**Comment by michael-hawker** [comment]
Another comment in #289 was about the extension also changing files that don't need to be changed.

I would imagine we'd also want to maybe add a check here:

https://github.com/Xavalon/XamlStyler/blob/a0c3bae2cead29bdd212b3bf5f046867e887eef5/src/XamlStyler.Extension.Windows.Shared/StylerPackage.Formatting.cs#L96-L103

And if the source matches the original that we don't call the `ReplaceText` call?

File: src/XamlStyler.Console/XamlStyler.Console.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <PackageId>XamlStyler.Console</PackageId>
-    <PackageVersion>3.1.0.0</PackageVersion>
+    <PackageVersion>3.1.1.0</PackageVersion>
```
Comment (michael-hawker) [inline src/XamlStyler.Console/XamlStyler.Console.csproj:4]:
```suggestion
    <PackageVersion>3.1.0.0</PackageVersion>
```

Revert change of build number as it's owned by the pipeline? Though I'd suggest to release changing to .NET 6 target as at least a `3.2` vs. a `3.1.x`?

Comment (grochocki) [inline src/XamlStyler.Console/XamlStyler.Console.csproj:4]: Nit: Can you revert this change? Our build pipeline will overwrite versioning for release.

Comment (grochocki) [reply src/XamlStyler.Console/XamlStyler.Console.csproj:4]: Actually, can you revert this change? Our build pipeline will overwrite versioning for release.

**Comment by grochocki** [comment]
> Another comment in #289 was about the extension also changing files that don't need to be changed.
> 
> I would imagine we'd also want to maybe add a check here:
> 
> https://github.com/Xavalon/XamlStyler/blob/a0c3bae2cead29bdd212b3bf5f046867e887eef5/src/XamlStyler.Extension.Windows.Shared/StylerPackage.Formatting.cs#L96-L103
> 
> And if the source matches the original that we don't call the `ReplaceText` call?

Yes, this is the right place to add the check. It will require a bit more testing with the VS extension, so we can spin this off as a separate issue.

**Comment by michael-hawker** [comment]
@grochocki I separated them all out as different commits originally. Did you just want to cherry-pick the commits you care about, or have me rebase without the commit for #289?

**Comment by grochocki** [comment]
Oh, no, let's keep the fix for #289 in here, and then I can open a new issue to fix in VSIX.

**Comment by michael-hawker** [comment]
Sounds good, so I think we're all set now with the commit of the revert for the version number? Let me know if there's anything else you need changed.

**Review by grochocki — APPROVED** [review]

**Comment by grochocki** [comment]
Thanks @michael-hawker!
